src/cogs/snipe.py

```diff
@@ -0,0 +1,107 @@
+from datetime import datetime
+from dataclasses import dataclass
+from typing import cast
+
+import discord
+from discord.ext import commands
+from discord.ext.commands import Context
+
+from src.util.blacklist import blacklist
+from src.util.snipe import *
+from src.util import util
+from src import config as conf
+
+@dataclass
+class MessageHistory:
+    deleted: datetime
+    states: list[discord.Message]
+
+class Sniper(commands.Cog, name="Snipe"):
+    def __init__(self, bot):
+        self.bot = bot
+
+        # Maps channel : list of history of deleted messages
+        self._deleted: dict[int, list[MessageHistory]] = {}
+
+        # Maps message id : history of edited message
+        self._message_history: dict[int, list[discord.Message]] = {}
+
+    @commands.hybrid_command(with_app_command=True)
+    async def snipe(self, ctx: Context, number: int = 0):
+        if ctx.channel.id not in self._deleted:
+            return await ctx.send("No message to snipe.")
+
+        histories = self._deleted[ctx.channel.id]
```

Comment:
misleading: This is one history of messages, not multiple histories. Did you mean `messages`?